src/main/java/com/aws/greengrass/shadowmanager/sync/model/CloudUpdateSyncRequest.java

```diff
@@ -172,21 +172,25 @@ public void execute(SyncContext context) throws RetryableException, SkipSyncRequ
     /**
      * Checks if it is necessary to perform an update to the cloud shadow.
      *
-     * @param context the execution context.
-     * @return true if cloud shadow update is necessary; Else false.
-     * @throws SkipSyncRequestException if the update request on the cloud shadow failed for another 400 exception.
-     * @throws UnknownShadowException   if the shadow sync information is missing
+     * @param context unused
+     * @return true
+     * @throws SkipSyncRequestException never
+     * @throws UnknownShadowException   never
      */
     @Override
     public boolean isUpdateNecessary(SyncContext context) throws SkipSyncRequestException, UnknownShadowException {
-        Optional<ShadowDocument> shadowDocument = context.getDao().getShadowThing(getThingName(), getShadowName());
-
-        //TODO: store this information in a return object to avoid unnecessary calls to DAO.
-        SyncInformation currentSyncInformation = context.getDao()
-                .getShadowSyncInformation(getThingName(), getShadowName())
-                .orElseThrow(() -> new UnknownShadowException("Shadow not found in sync table"));
-
-        return isUpdateNecessary(shadowDocument, currentSyncInformation, context);
+        // This method is called in context of an IPC-handling thread.
```

Comment:
Where is this method called from the IPC handling thread?

Reply:
here's the sequence of calls:

https://github.com/aws-greengrass/aws-greengrass-shadow-manager/blob/8e24c7de2f5d091fb6a42247742d28be76f96a5a/src/main/java/com/aws/greengrass/shadowmanager/ipc/UpdateThingShadowRequestHandler.java#L253

https://github.com/aws-greengrass/aws-greengrass-shadow-manager/blob/8e24c7de2f5d091fb6a42247742d28be76f96a5a/src/main/java/com/aws/greengrass/shadowmanager/sync/SyncHandler.java#L246

https://github.com/aws-greengrass/aws-greengrass-shadow-manager/blob/8e24c7de2f5d091fb6a42247742d28be76f96a5a/src/main/java/com/aws/greengrass/shadowmanager/sync/strategy/BaseSyncStrategy.java#L280

https://github.com/aws-greengrass/aws-greengrass-shadow-manager/blob/8e24c7de2f5d091fb6a42247742d28be76f96a5a/src/main/java/com/aws/greengrass/shadowmanager/sync/model/CloudUpdateSyncRequest.java#L189

https://github.com/aws-greengrass/aws-greengrass-shadow-manager/blob/8e24c7de2f5d091fb6a42247742d28be76f96a5a/src/main/java/com/aws/greengrass/shadowmanager/sync/model/CloudUpdateSyncRequest.java#L212

Reply:
The `BaseSyncStrategy` should happen in a separate thread. It should not be a sequential call since `putSyncRequest` just puts the request in the sync queue. We have a separate function which executes these sync requests.

Reply:
Spoke offline. Change looks good. Add an integ test to verify this change.

Reply:
I believe we need to block on the shadow name. So have a central lock system based on the shadow name.

Reply:
right. it would block on the shadow. I wasn't thinking of a global lock

Reply:
for the case where you a syncing a single shadow with constant updates, it _could_ slow that down, but it may be a better overall solution

Reply:
Right. A central locking system based on the shadow name might delay it by a few ms. But I also think that its a better overall solution.